txpipe/twopoint.py

```diff
@@ -399,23 +402,24 @@ def write_output(self, source_list, lens_list, meta, results):
 
         # Load the tracer data N(z) from an input file and
         # copy it to the output, for convenience
-        if source_list:
-            with self.open_input("shear_photoz_stack", wrapper=True) as f:
-                for i in source_list:
-                    z, Nz = f.get_bin_n_of_z(i)
-                    S.add_tracer("NZ", f"source_{i}", z, Nz)
-                    if self.config["do_shear_pos"] == True:
-                        S2.add_tracer("NZ", f"source_{i}", z, Nz)
-
-
-        if lens_list:
-            with self.open_input("lens_photoz_stack", wrapper=True) as f:
-                # For both source and lens
-                for i in lens_list:
-                    z, Nz = f.get_bin_n_of_z(i)
-                    S.add_tracer("NZ", f"lens_{i}", z, Nz)
-                    if self.config["do_shear_pos"] == True:
-                        S2.add_tracer("NZ", f"lens_{i}", z, Nz)
+        if self.config["do_shear_pos"] or self.config["do_shear_shear"]:
+            if source_list:
+                with self.open_input("shear_photoz_stack", wrapper=True) as f:
+                    for i in source_list:
+                        z, Nz = f.get_bin_n_of_z(i)
+                        S.add_tracer("NZ", f"source_{i}", z, Nz)
+                        if self.config["do_shear_pos"] == True:
+                            S2.add_tracer("NZ", f"source_{i}", z, Nz)
+
+        if self.config["do_pos_pos"] or self.config["do_shear_pos"]:
+            if lens_list:
+                with self.open_input("lens_photoz_stack", wrapper=True) as f:
+                    # For both source and lens
+                    for i in lens_list:
+                        z, Nz = f.get_bin_n_of_z(i)
+                        S.add_tracer("NZ", f"lens_{i}", z, Nz)
+                        if self.config["do_shear_pos"] == True:
+                            S2.add_tracer("NZ", f"lens_{i}", z, Nz)
```

Comment:
Having two if statements here means e.g. if `do_shear_shear` is True, but no source bins were defined we don't throw an error, the two-point file just wont contain shear-shear correlations even though it was asked for. I think thats fine, just wanted to point it out and make sure everyone agreed that's desired

Reply:
I just added an else statement to throw an error for this.